crates/sui-e2e-tests/tests/shared_objects_version_tests.rs

```diff
@@ -263,22 +224,24 @@ impl TestEnvironment {
             )
             .await?;
 
-        assert!(fx.status().is_ok());
-
-        Ok(*fx
+        let mutated = fx
             .mutated()
             .iter()
-            .find(|(obj, _)| obj.0 == counter)
-            .expect("Counter modified"))
+            .find(|owner_ref| owner_ref.reference.object_id == counter)
+            .expect("Counter modified");
+        Ok((mutated.reference.to_object_ref(), mutated.owner))
     }
 }
 
-async fn publish_move_package(
-    gas: Object,
-    net_addresses: &[Multiaddr],
-    gas_price: u64,
-) -> ObjectRef {
+async fn publish_move_package(test_cluster: &TestCluster) -> ObjectRef {
     let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
     path.push("tests/move_test_code");
-    publish_package(gas, path, net_addresses, gas_price).await
+    test_cluster.wallet.publish_package(path).await
+}
+
+fn assert_move_abort_error(error: &str, package: &AccountAddress, module_name: &str, code: u64) {
```

Comment:
This is not quite ideal and I could take a look at exposing raw results from test cluster